src/Sarif/FileRegionsCache.cs

```diff
@@ -92,6 +105,158 @@
+            bool lineStart = true;
+            // Was the previous character a CR (carriage return)
+            bool prevCR = false;
+
+            Dictionary<string, int> hashCounts = new Dictionary<string, int>();
+
+            // Output the current hash and line number to the cache
+            Action outputHash = () =>
+            {
+                ulong uhashRaw = (ulong)hashRaw;
+                string hashValue = uhashRaw.ToString("x16");
+
+                if (!hashCounts.ContainsKey(hashValue))
+                {
+                    hashCounts[hashValue] = 0;
+                }
+
+                hashCounts[hashValue]++;
+                _rollingHashesCache[filePath][lineNumbers[index]] = $"{hashValue}:{hashCounts[hashValue]}";
+                lineNumbers[index] = -1;
+            };
+
+            // Update the current hash value and increment the index in the window
+            Action<int> updateHash = (current) =>
+            {
+                int begin = window[index];
+                window[index] = current;
+
+                hashRaw = (MOD * hashRaw) + (long)current - (firstMod * (long)begin);
+
+                index = (index + 1) % BLOCK_SIZE;
+            };
+
+            // First process every character in the input, updating the hash and lineNumbers
+            // as we go. Once we reach a point in the window again then we've processed
+            // BLOCK_SIZE characters and if the last character at this point in the window
+            // was the start of a line then we should output the hash for that line.
+            Action<int> processCharacter = (current) =>
+            {
+                // skip tabs, spaces, and line feeds that come directly after a carriage return
+                if (current == space || current == tab || (prevCR && current == lf))
+                {
+                    prevCR = false;
+                    return;
+                }
+                // replace CR with LF
+                if (current == cr)
+                {
+                    current = lf;
+                    prevCR = true;
+                }
+                else
+                {
+                    prevCR = false;
+                }
+                if (lineNumbers[index] != -1)
+                {
+                    outputHash();
+                }
+                if (lineStart)
+                {
+                    lineStart = false;
+                    lineNumber++;
+                    lineNumbers[index] = lineNumber;
+                }
+                if (current == lf)
+                {
+                    lineStart = true;
+                }
+                updateHash(current);
+            };
+
+            string fileText = null;
+            try
+            {
+                if (_fileSystem.FileExists(filePath))
+                {
+                    fileText = _fileSystem.FileReadAllText(filePath);
+                }
+            }
+            catch (IOException) { }
```

Comment:
>IOException [](http://example.com/codeflow?start=19&length=11)

?? i don't suggest swallowing this.

Reply:
@michaelcfanning Thanks for pointing this out. Will follow your pattern as we discussed earlier today. Will raise a GitHub issue subsequently to track this.